pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ComparisonColumns.java

```diff
@@ -52,18 +52,9 @@ public int compareToSealed(ComparisonColumns other) {
         continue;
       }
 
-      // Always keep the record with non-null value, or that with the greater comparisonResult
-      if (comparisonValue == null) {
-        // implies comparisonValue == null && otherComparisonValue != null
-        return -1;
-      } else if (otherComparisonValue == null) {
-        // implies comparisonValue != null && otherComparisonValue == null
-        return 1;
-      } else {
-        int comparisonResult = comparisonValue.compareTo(otherComparisonValue);
-        if (comparisonResult != 0) {
-          return comparisonResult;
-        }
+      int comparisonResult = comparisonValue.compareTo(otherComparisonValue);
```

Comment:
no more null-checking required, because _values will always hold all non-null entries (where some may be `<defaultNullValue>`